### PR TITLE
Add some facts about decidability of partial orders/equivalences.

### PR DESCRIPTION
**Pull request by sstucki**

### CODE REVIEW

File: src/Data/Fin/Properties.agda [hunk 1]
```diff
@@ -1008,8 +1008,25 @@ module _ {f} {F : Set f → Set f} (RF : RawFunctor F) where
 -- If there is an injection from a type A to a finite set, then the type
 -- has decidable equality.
 
+module FromInjection {n a e} {A : Setoid a e}
+                     (inj : Injection A (≡-setoid n)) where
```
Comment (MatthewDaggitt) [inline src/Data/Fin/Properties.agda:1012]: Hmm so we don't usually use named inner modules, and if we do we always open them publicly afterwards. The definition of `FromInjection.eq?` is compatible with `eq` right?  `eq?` is also a terrible name for the existing property. So how about we deprecate `eq?` and then add the following:

```agda
module _ {ℓ} {S : Setoid a ℓ} (inj : Injection S (≡-setoid n)) where
  
  inj⇒≟ : -> B.Decidable _≈_
  inj⇒≟ inj = Dec.via-injection inj _≟_

  inj⇒decSetoid : DecSetoid a e
  inj⇒decSetoid = ...
``

File: src/Relation/Binary/Lattice/Properties/JoinSemilattice.agda [hunk 2]
```diff
@@ -101,3 +107,18 @@ dualMeetSemilattice = record
...
+
+------------------------------------------------------------------------
+-- If ≈ is decidable then so is ≤
+
+≈-dec⇒≤-dec : Decidable _≈_ → Decidable _≤_
+≈-dec⇒≤-dec _≟_ x y with (x ∨ y) ≟ y
+... | yes x∨y≈y = yes (trans (x≤x∨y x y) (reflexive x∨y≈y))
+... | no  x∨y≉y = no (contraposition x≤y⇒x∨y≈y x∨y≉y)
+
+≈-dec⇒dec : Decidable _≈_ → IsDecPartialOrder _≈_ _≤_
```
Comment (MatthewDaggitt) [inline src/Relation/Binary/Lattice/Properties/JoinSemilattice.agda:119]: I know it's a bit verbose, but I think the names would be a bit more meaningful if we changed this (and the corresponding ones in the other modules) to `≈-dec⇒isDecPartialOrder`

**Review by MatthewDaggitt — COMMENTED** [review]
Thanks for the PR! Other than those two comments, I think we just need some entries in `CHANGELOG.md`.

File: src/Data/Fin/Properties.agda (hunk 1 above)
Comment (sstucki) [reply src/Data/Fin/Properties.agda]: Done.

File: src/Relation/Binary/Lattice/Properties/JoinSemilattice.agda (hunk 2 above)
Comment (sstucki) [reply src/Relation/Binary/Lattice/Properties/JoinSemilattice.agda]: Done.

**Comment by sstucki** [comment]
Made all the suggested changes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -489,7 +489,12 @@ Non-backwards compatible changes
   * `fold-*`
   * `fold-pull`
 
-* In `Data.Fin.Properties` the `i` argument to `opposite-suc` has been made explicit.
+* In `Data.Fin.Properties`:
+  + the `i` argument to `opposite-suc` has been made explicit;
+  + the proof that an injection from a type `A` into `Fin n` induces a
+    decision procedure for `_≡_` on `A` has been generalized to other
+    equivalences over `A` (i.e. to arbitrary setoids), and renamed
+    from `eq?` to the more descriptive `inj⇒≟` and `inj⇒decSetoid`.
```
Comment (MatthewDaggitt) [inline CHANGELOG.md:497]: Sorry another niggly organisation point. The generalisation is backwards compatible so should go under `Other minor changes`, the deprecation of `eq?` should live under the `Deprecated names` section. Other than that all looks good.

Comment (sstucki) [reply CHANGELOG.md]: Fixed. I hope it's as it should be now. 😄

Comment (MatthewDaggitt) [reply CHANGELOG.md]: Yes, thanks!

**Comment by MatthewDaggitt** [comment]
Merging in, thanks again for the PR!

**Pull request merged**